pkgs/build-support/setup-hooks/patch-dylib.sh

```diff
@@ -0,0 +1,59 @@
+# The equivalent of patchelf but for .dylib / mach-o / shared libraries on macOS
+#
+# Patches the paths to shared library dependencies to use libs in the nix store.
+# The script is not magical and expects the caller to provide a list of paths
+# where library dependencies can be found.
+#
+# For example if a.dylib depended on /usr/lib/b.dylib, there's a chance the latter
+# does not actually exist and can be found somewhere in /nix/store/XXX-lib/lib/b.dylib.
+#
+# Dependencies:
+#  - grep
+#  - sed (probably in the gnused package)
+#
+# Arguments:
+#  1. targetLib: path to a shared library that has to be patched
+#  2. libraryPaths: <colon> (:) separated list of paths
+function patchDylib(){
+  local targetLib="$1"
+  local libraryPaths="$2"
+
+  # Make the shared paths absolute
+  install_name_tool -id $targetLib $targetLib
+  # Get all dependent shared libs
+  libs="$(otool -L $targetLib \
+          | sed --regexp-extended \
+                --expression='s|^\s+(.+)\s\(.+\)|\1|g' \
+          | grep --extended-regexp \
+                 --invert-match '^(/nix/store|/System)'
```

Comment:
```suggestion
                 --invert-match '^(/nix/store|/System|/usr/lib/libSystem)'
```
I think this should solve the issue with libSystem, but I haven't tested it, so I'm not completely sure